addons/hr_work_entry_contract_attendance/__init__.py

```diff
@@ -0,0 +1,85 @@
+#-*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
+from odoo import api, SUPERUSER_ID
+from . import models
+from . import wizard
+
+def _generate_attendances(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+
+    hne_contract = env.ref('hr_work_entry_contract_attendance.hr_contract_hne', raise_if_not_found=False)
+    if hne_contract:
+        if not env['hr.attendance'].sudo().search_count([('employee_id', '=', hne_contract.employee_id.id)]):
+            employee = hne_contract.employee_id
+            #generate attendances for ngh for every day up until last week (those will be hardcoded with weird hours)
+            today = datetime.today()
+            delta = (today + relativedelta(weeks=-1, days=-today.weekday(), weekday=0)).date() - hne_contract.date_start
+            attendance_create_vals = []
+            for i in range(delta.days):
+                day = hne_contract.date_start + timedelta(days=i)
+                #not on weekend days
+                if day.weekday() >= 5:
+                    continue
+                attendance_create_vals.extend([
+                    {
+                        'employee_id': employee.id,
+                        'check_in': day.strftime('%Y-%m-%d 08:00:00'),
+                        'check_out': day.strftime('%Y-%m-%d 12:00:00'),
+                    },
+                    {
+                        'employee_id': employee.id,
+                        'check_in': day.strftime('%Y-%m-%d 13:00:00'),
+                        'check_out': day.strftime('%Y-%m-%d 17:00:00'),
+                    }
+                ])
+            attendance_create_vals.extend([
+                {
+                    "employee_id": employee.id,
+                    "check_in": (today+relativedelta(weeks=-1, days=-today.weekday(), weekday=0)).strftime('%Y-%m-%d 08:00:00'),
```

Comment:
The ORM accepts datetime objects:
```py
                    "check_in": today + relativedelta(weeks=-1, days=-today.weekday(), weekday=0),
```

Reply:
Actually while it might be confusing I was using this to set the hours on the attendance `%Y-%m-%d 08:00:00`